trinity/plugins/builtin/light_peer_chain_bridge/light_peer_chain_bridge.py

```diff
@@ -62,36 +137,80 @@ class LightPeerChainEventBusResponder:
     def __init__(self, chain: BaseLightPeerChain, event_bus: Endpoint) -> None:
         self.chain = chain
         self.event_bus = event_bus
-        asyncio.ensure_future(self.answer_requests())
-
-    async def answer_requests(self) -> None:
-        async for event in self.event_bus.stream(LightPeerChainRequest):
-
-            method = getattr(self.chain, event.method_name)
-
-            if not callable(method):
-                self.event_bus.broadcast(
-                    LightPeerChainResponse(Exception(f"Not a method: {event.method_name}")),
-                    event.broadcast_config()
-                )
-                continue
-
-            try:
-                if event.payload is not None:
-                    response = await method(*event.payload)
-                else:
-                    response = await method()
-            except Exception as e:
-                # we send the exception over to re-raise it on the consumer side (other process)
-                self.event_bus.broadcast(
-                    LightPeerChainResponse(e),
-                    event.broadcast_config()
-                )
-            else:
-                self.event_bus.broadcast(
-                    LightPeerChainResponse(response),
-                    event.broadcast_config()
-                )
+        asyncio.ensure_future(self.answer_get_blockheader_by_hash_requests())
+        asyncio.ensure_future(self.answer_get_blockbody_by_hash_requests())
+        asyncio.ensure_future(self.answer_get_receipts_by_hash_requests())
+        asyncio.ensure_future(self.answer_get_account_requests())
+        asyncio.ensure_future(self.answer_get_contract_code_requests())
+
+    async def answer_get_blockheader_by_hash_requests(self) -> None:
+        async for event in self.event_bus.stream(GetBlockHeaderByHashRequest):
+
+            val, error = await self._pass_or_raise(
+                self.chain.get_block_header_by_hash(event.block_hash)
+            )
+
+            self.event_bus.broadcast(
+                BlockHeaderResponse(val, error),
+                event.broadcast_config()
+            )
+
+    async def answer_get_blockbody_by_hash_requests(self) -> None:
+        async for event in self.event_bus.stream(GetBlockBodyByHashRequest):
+
+            val, error = await self._pass_or_raise(
+                self.chain.get_block_body_by_hash(event.block_hash)
+            )
```

Comment:
Even though `mypy` knows that `val` is a `BlockBody`, there is still some room for human errors in here. This is unrelated to this PR and boils down to the fact that e.g. `BlockBody() + BlockHeader()` is perfectly fine from `mypy`'s perspective. I guess that is because both can be serialized to `bytes`?

So, that means there is room for errors where you e.g. subscribe to `GetBlockBodyByHashRequests` but then feed a `BlockHeader` into that response.